boxsdk/object/folder.py

```diff
@@ -392,7 +392,7 @@ def update_sync_state(self, sync_state):
         return self.update_info(data=data)
 
     @api_call
-    def add_collaborator(self, collaborator, role, notify=False):
+    def add_collaborator(self, collaborator, role, notify=False, can_view_path=False):
```

Comment:
Two things I wonder about this:

1) Can we safely put this before `notify`? It'd be nice to put all of the object params before the `notify` param, since the latter isn't really related to business logic. The question is, how many people use `notify`, and how many of those use it as a positional argument? Even if we're worried about that, we can still make it as a breaking change, since we're in the transition to 2.0.0.

2) Rather than stacking a bunch of optional params at the end (it's only `can_view_path` for now, but who knows, in the future there could be others), I wonder if we should instead expose a generic parameter for extra body params. Something like `(additional | extra | optional)_attributes`. And then callers could do `additional_attributes={'can_view_path': True, 'foo': 'bar'}`

Thoughts @Jeff-Meadows

Reply:
1. I'm not sure why `notify` isn't business logic. The SDK doesn't get to specify what is or isn't business logic. Sending or not emails could very much be business logic???

That said I'm fine placing this param anywhere.

2. I thought about that option and then went to the Docs and saw that can_view_path was the only unsupported param.

But thinking long term / broadly... having some sort of policy/pattern in the SDK's API that maps nicely to the actual V2 API would be nice. Something like::
* required params have explicit named-params.
* optionals params are supported via the optional dicts.

Jordan - are you suggesting just body-params would have a dict parameter? What about query params?

Looking at https://developer.box.com/v2.0/reference#add-a-collaboration
* query param `fields`. its optional. The box-sdk doesn't support it.
* query param `notify` the docs don't list it's default. But I *guess* it's optional.
* body param `accessible_by`. It's not labeled as Required. Is that correct
* body param `can_view_path`. Optional, so it could be cover by a dict param

Thus the current boxSDK API "breaks" the model of optional being supported via dicts. So what then... it's just random. Some params are explicit. Others are supported via dict? Maybe that is OK. Maybe not. Or maybe in the move to 2.0 would implement a nice pattern like this? That would mean removing the `notify` param and having a "option_query_params" dict parameter.

Reply:
Jeff and I discussed this and agreed that the discussion / implementation of a pattern for supporting all current and future query-params and body params is beyond the scope of this change. And that this change as-is is ok.